src/Middleware.php

```diff
@@ -24,7 +24,7 @@ public static function generateSignature($host, $method, $path, $query, $date, $
             $query,
             $date,
             $idempotency,
-            hash('sha256', $body)
+            hash('sha256', ($body == null) ? "" : $body)
```

Comment:
This should archive the same and would be much shorter:
```suggestion
            hash('sha256', $body ?? "")
```

Below some tests:
```php
php > print_r( $body ?? 'fallback' );
fallback
```

```php
php > $body = null; print_r( $body ?? 'fallback' );
fallback
```

```php
php > $body = ''; print_r( $body ?? 'fallback' );
php >
```

```php
php > $body = 'some body'; print_r( $body ?? 'fallback' );
some body
```